### PR TITLE
docs: Update CONTRIBUTING.md with DCO

### LINKED ISSUE #2458: Include DCO instructions in Contributing Guide
While PRs are subject to a DCO signoff procedure (since #1388), this is not included in the Contributing Guide.

It looks like it should be a part of the steps provided in https://github.com/thanos-io/thanos/blob/master/CONTRIBUTING.md#pull-request-process but there may be other places it makes sense.

A great solution to this issue would include:
- actual command to run to sign off on commit changes
- explanation of what this means to the project
- a link to more information from the CNCF

### PR DESCRIPTION
**Pull request by ranjithkumar007**
Fixes #2458 

Signed-off-by: ranjithkumar007 <ranjith.dakshana2015@gmail.com>

* [ ] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

## Changes
Added DCO instructions in CONTRIBUTING.md

## Verification
Verified locally using `make web-serve` and on [deploy preview](https://deploy-preview-2465--thanos-io.netlify.app/contributing/#pull-request-process)

### CODE REVIEW

**Comment by ranjithkumar007** [comment]
Great documentation especially about the [design](https://thanos.io/design.md/) and the [components](https://thanos.io/components/) of Thanos. Enjoyed going through these.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -109,6 +109,19 @@ $ make build
 $ ./thanos -h
 ```
 
+1. Signing your work(Developer Certificate of Origin)
```
Comment (bwplotka) [inline CONTRIBUTING.md:112]:
```suggestion
1. Signing your work: DCO (Developer Certificate of Origin) Process.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -109,6 +109,19 @@ $ make build
...
+
+By contributing to this project you agree to the [Developer Certificate of Origin](https://developercertificate.org/)(DCO). This document was created by the Linux Kernel community and is a simple statement that you, as a contributor, have the legal right to make the contribution.
```
Comment (bwplotka) [inline CONTRIBUTING.md:114]: nice! :+1:

File: CONTRIBUTING.md (hunk 1 above)
Comment (ranjithkumar007) [reply CONTRIBUTING.md]: Done!

**Pull request merged**